conftest.py

```diff
@@ -73,3 +84,174 @@ def undo(self):
+                ret = 0
+            except subprocess.CalledProcessError as e:
+                output = e.output
+                ret = e.returncode
+            except SystemExit as e:  # possibly raised by argparse
+                output = ""
+                ret = e.code
+            if binary_output:
+                return ret, output
+            else:
+                return ret, os.fsdecode(output)
+        else:
+            stdin, stdout, stderr = sys.stdin, sys.stdout, sys.stderr
+            try:
+                sys.stdin = StringIO(input.decode())
+                sys.stdin.buffer = BytesIO(input)
+                output = BytesIO()
+                # Always use utf-8 here, to .decode() below
+                output_text = sys.stdout = sys.stderr = io.TextIOWrapper(output, encoding="utf-8")
+                if archiver is None:
+                    archiver = Archiver()
+                archiver.prerun_checks = lambda *args: None
+                archiver.exit_code = EXIT_SUCCESS
+                helpers.exit_code = EXIT_SUCCESS
+                try:
+                    args = archiver.parse_args(list(args))
+                    # argparse parsing may raise SystemExit when the command line is bad or
+                    # actions that abort early (eg. --help) where given. Catch this and return
+                    # the error code as-if we invoked a Borg binary.
+                except SystemExit as e:
+                    output_text.flush()
+                    return e.code, output.getvalue() if binary_output else output.getvalue().decode()
+                try:
+                    ret = archiver.run(args)  # calls setup_logging internally
+                finally:
+                    flush_logging()  # usually done via at exit, but we do not exit here
+                output_text.flush()
+                return ret, output.getvalue() if binary_output else output.getvalue().decode()
+            finally:
+                sys.stdin, sys.stdout, sys.stderr = stdin, stdout, stderr
+
+    return cmd
+
+
+@pytest.fixture()
+def create_src_archive(cmd_fixture, archiver_setup):
+    repo_location = archiver_setup.repository_location
+    source_dir = archiver_setup.src_dir
+
+    def src_archive(name, ts=None):
+        if ts:
+            cmd_fixture(f"--repo={repo_location}", "create", "--compression=lz4", f"--timestamp={ts}", name, source_dir)
+        else:
+            cmd_fixture(f"--repo={repo_location}", "create", "--compression=lz4", name, source_dir)
+
+    return src_archive
+
+
+@pytest.fixture()
+def open_archive(archiver_setup):
+    def archive_and_repository(name):
+        repo_path = archiver_setup.repository_path
+        repository = Repository(repo_path, exclusive=True, create=True)
+        with repository:
+            manifest = Manifest.load(repository, Manifest.NO_OPERATION_CHECK)
+            archive = Archive(manifest, name)
+        return archive, repository
```

Comment:
if you do it like that, leaving the context will close the repository before you return the repository (and the archive).

Reply:
Good point. I will rethink this approach.

Reply:
After looking more into this, it appears closing the repo and archive and returning them is the intended behavior. The tests that use this tend to then use a CM `with repository` and do the actions needed.

Reply:
OK. Also: this is not a change, it behaved like that already before this PR.